packages/common-all/src/error.ts

```diff
@@ -45,15 +44,64 @@ type ServerErrorProps = {
   code?: StatusCodes;
 };
 
-export type IDendronError = DendronErrorProps;
+export type IDendronError<TCode = StatusCodes | undefined> =
+  DendronErrorProps<TCode>;
 
-export class DendronError extends Error implements IDendronError {
+export class DendronError<TCode = StatusCodes | undefined>
+  extends Error
+  implements IDendronError<TCode>
+{
   public status?: string;
   public payload?: string;
   public severity?: ERROR_SEVERITY;
-  public code?: number;
+  public code?: TCode;
   public innerError?: Error;
 
+  /** The output that may be displayed to a person if they encounter this error. */
+  public stringifyForHumanReading() {
+    return this.message;
+  }
+
+  /** Overload this to change how the `payload` is stringified. */
+  protected payloadStringify() {
+    return JSON.stringify(this.payload);
+  }
+
+  /** The output that may be saved into the local logs for the user. */
+  public stringifyForLogs() {
+    const { severity, code, message } = this;
+    const payload: { [key: string]: any } = {
+      severity,
+      code,
+      message,
+    };
+    if (this.innerError) {
+      payload.innerError = this.innerError;
+    }
+    if (this.payload) {
+      payload.payload = this.payloadStringify();
+    }
+    return JSON.stringify(payload);
+  }
+
+  /** The output that may be sent to Sentry, or other telemetry service.
+   *
+   * This function will eventually check that the output is stripped of PII,
+   * but for now that's the same as these.
+   */
+  public stringifyForTelemetry() {
```

Comment:
_[My bad - I had this comment in the review earlier but it somehow got deleted]_

nit: can we tweak this description slightly?

 /** The output that may be sent to Sentry, or other telemetry service.
   *
   * This return value from this function should be stripped of PII - if your derived error class has PII in its payload, then override this function and provide a payload that has the PII removed.
   */
  public stringifyForTelemetry() {
    // This function will eventually try to do its own checks to strip out PII, but for now the default is to return the same string as `stringifyForLogs()`
    return this.stringifyForLogs();
  }